java/src/com/here/flexpolyline/PolylineEncoderDecoder.java

```diff
@@ -213,9 +213,9 @@ private boolean hasThirdDimension() {
         private void decodeHeader() {
             AtomicLong header = new AtomicLong(0);
             decodeHeaderFromString(encoded, index, header);
-            precision = (int) (header.get() & 15); // we pick the first 3 bits only
+            precision = (int) (header.get() & 15); // we pick the first 4 bits only
```

Comment:
Good catch! :) But please open a separate PR for this so we can keep the number of affected implementations down to one per PR.